### PR TITLE
adding a max retry attempt check for each message

### PR DESCRIPTION
**Pull request by shivasishdas**

### CODE REVIEW

File: internal/subscriber/subscriber.go [hunk 1]
```diff
@@ -88,6 +89,8 @@ func (s *Subscriber) GetSubscription() string {
 
 // commits existing message on primary topic and pushes message to the pre-defined retry topic
 func (s *Subscriber) retry(ctx context.Context, retryMsg *RetryMessage) {
+	retryMsg.incrementRetry()
```
Comment (Abhishekvrshny) [inline internal/subscriber/subscriber.go:92]: should this be incremented here or after the check https://github.com/razorpay/metro/pull/89/files#diff-4bcad8e79804b094286e6698fe58e6f08d402b61a217a004fd4b078034ee9992R107?

File: pkg/messagebroker/kafka.go [hunk 2]
```diff
@@ -351,17 +359,28 @@ func (k *KafkaBroker) ReceiveMessages(ctx context.Context, request GetMessagesFr
...
 	for {
 		var msgID string
+		var retryCounter int32
 		msg, err := k.Consumer.ReadMessage(time.Duration(request.TimeoutSec) * time.Second)
 		if err == nil {
 			for _, v := range msg.Headers {
 				if v.Key == messageID {
 					msgID = string(v.Value)
+				} else if v.Key == retryCount {
+					rc, _ := strconv.ParseInt(string(v.Value), 10, 0)
```
Comment (Abhishekvrshny) [inline pkg/messagebroker/kafka.go:369]: data was serialised with json marshal earlier. Follow 1 convention

Comment (shivasishdas) [reply pkg/messagebroker/kafka.go]: changed

File: internal/subscriber/subscriber.go (hunk 1 above)
Comment (shivasishdas) [reply internal/subscriber/subscriber.go]: changed

**Review by Abhishekvrshny — APPROVED** [review]

**Pull request merged**